Makefile

```diff
@@ -42,11 +42,15 @@ tinycbor/Makefile crypto/tiny-AES-c/aes.c:
 .PHONY: cbor
 cbor: $(LIBCBOR)
 
-$(LIBCBOR): tinycbor/Makefile
+$(LIBCBOR):
 	cd tinycbor/ && $(MAKE) clean && $(MAKE) -j8
 
-test:
+test: env3
+	$(MAKE) clean
 	$(MAKE) -C . main
+	$(MAKE) clean
+	$(MAKE) -C ./targets/stm32l432 test PREFIX=$(PREFIX) "VENV=$(VENV)"
+	$(MAKE) clean
```

Comment:
I don't understand all the "make clean" above.   I think I'd rewrite the entire target as:
```
test: main
    $(MAKE) -C ./targets/stm32l432 test PREFIX=$(PREFIX) "VENV=$(VENV)"
```

Reply:
What's changing the architecture so that line 49/50 build with arm instead of x86?

Reply:
not the build architecture, the target architecture. the PC / main is building for x86 and the stm32l432 is building for arm with a arm targeted toolchain, as can be seen in the stm32l432 sub-makefiles